exercises/views.py

```diff
@@ -121,6 +125,10 @@ def submit_autograding_exercise(request, exercise):
         redirect_url = reverse('exercise_detail', kwargs={'exercise_tag': exercise.tag})
         if not is_solution:
             redirect_url += '?wrong_answer=1'
+            # Send the error message returned from the grader
+            if error_msg:
+                redirect_url += '&error_msg=' + error_msg
```

Comment:
This way of passing error messages looks OK. See also the possibility of using django helpers:

https://docs.djangoproject.com/en/1.9/topics/http/shortcuts/#redirect